src/snapred/backend/recipe/algorithm/UnfocusedNormalizationCorrection.py

```diff
@@ -0,0 +1,209 @@
+import json
+from typing import Any, Dict, Tuple
+
+import numpy as np
+from mantid.api import (
+    AlgorithmFactory,
+    PythonAlgorithm,
+)
+from mantid.kernel import Direction
+from scipy.interpolate import make_smoothing_spline, splev
+
+from snapred.backend.dao.ingredients import ReductionIngredients as Ingredients
+from snapred.backend.dao.state.CalibrantSample.CalibrantSamples import CalibrantSamples
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "UnfocusedNormalizationCorrection"
+
+
+# TODO: Rename so it matches filename
+class UnfocusedNormalizationCorrection(PythonAlgorithm):
+    def PyInit(self):
+        # declare properties
+        self.declareProperty("Ingredients", defaultValue="", direction=Direction.Input)
+        self.declareProperty("CalibrantSample", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspace", defaultValue="vanadiumrawcorr_out", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def chopIngredients(self, ingredients: Ingredients) -> None:
+        stateConfig = ingredients.reductionState.stateConfig
+        self.liteMode: bool = stateConfig.isLiteMode
+        self.vanadiumRunNumber: str = ingredients.runConfig.runNumber
+        self.vanadiumBackgroundRunNumber: str = stateConfig.emptyInstrumentRunNumber
+        self.geomCalibFile: str = stateConfig.geometryCalibrationFileName
+        # iPrm['calibrationDirectory'] + sPrm['stateID'] +'/057514/'+ f'RVMB{VRun}
+        self.rawVFile: str = stateConfig.rawVanadiumCorrectionFileName
```

Comment:
Might be an issue here.

Reply:
Okay, found it. This file is defined within src.tests.resources.inputs.ReductionParameters.json -> rawVCorrFileName. Unsure as to why this is being invoked at all.